client/cli/src/logger.rs

```diff
@@ -0,0 +1,270 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2020 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use ansi_term::Colour;
+use flexi_logger::{
+    DeferredNow, Duplicate, LogSpecBuilder,
+    LogSpecification, LogTarget, Logger, Criterion, Naming, Cleanup, Age,
+};
+use lazy_static::lazy_static;
+use regex::Regex;
+use std::path::PathBuf;
+use structopt::{
+	StructOpt,
+};
+use crate::error::{Error, Result};
+
+type IoResult = std::result::Result<(), std::io::Error>;
+
+/// Default size used for rotation. Corresponds to 100 MiB.
+const DEFAULT_ROTATION_SIZE: u64 = 104_857_600;
+
+/// Options for log rotation.
+#[derive(Debug, StructOpt)]
+pub struct LogRotationOpt {
+	/// Specify the path of the directory which will contain the log files.
+	/// Defaults to rotating logs once they reach 100 MiB.
+	#[structopt(long, parse(from_os_str))]
+	log_directory: Option<PathBuf>,
+	
+	/// Rotate the log file when it has become older than the specified age.
+	#[structopt(long,
+		conflicts_with("log-size"), 
+		possible_values(&["day", "hour", "minute", "second"]), 
```

Comment:
So you'd write `--log-age minute`? I don't understand this.

Reply:
Yes, it's inspired by the `Age`[https://docs.rs/flexi_logger/0.15.7/flexi_logger/enum.Age.html] enum from flexi_logger :) I could remove minute / second or even rename them if you think we could use a better name ! :)

Reply:
The documentation of the `Age` enum that you linked does make sense because it explains what the values mean.

Reply:
https://github.com/paritytech/substrate/pull/6564/commits/5417943b26ee38273e05efe4d7f19d47fa6a5a26 :)